spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java

```diff
@@ -166,6 +186,137 @@ BraveHttpClientHandler braveHttpClientHandler(
+		static class BraveBaggageConfiguration {
+
+			@Bean
+			@ConditionalOnMissingBean
+			@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "W3C", matchIfMissing = true)
+			BaggagePropagation.FactoryBuilder w3cPropagationFactory(BraveBaggageManager braveBaggageManager, TracingProperties tracingProperties) {
+				return new W3CPropagation(braveBaggageManager, tracingProperties.getBaggage().getLocalFields());
+			}
+
+			@Bean
+			@ConditionalOnMissingBean
+			@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "B3", matchIfMissing = true)
+			BaggagePropagation.FactoryBuilder b3PropagationFactory() {
+				return BaggagePropagation.newFactoryBuilder(B3Propagation.newFactoryBuilder().injectFormat(B3Propagation.Format.SINGLE_NO_PARENT).build());
+			}
+
+			@Bean
+			@ConditionalOnMissingBean
+			@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "AWS")
+			@ConditionalOnClass(AWSPropagation.class)
+			BaggagePropagation.FactoryBuilder awsPropagationFactory() {
+				return BaggagePropagation.newFactoryBuilder(AWSPropagation.FACTORY);
+			}
+
+			@Bean
+			@ConditionalOnMissingBean
+			Propagation.Factory micrometerTracingPropagationWithB3Baggage(BaggagePropagation.FactoryBuilder factoryBuilder,
+					TracingProperties tracingProperties, ObjectProvider<List<BaggagePropagationCustomizer>> baggagePropagationCustomizers) {
+				List<String> localFields = tracingProperties.getBaggage().getLocalFields();
+				for (String fieldName : localFields) {
+					factoryBuilder.add(BaggagePropagationConfig.SingleBaggageField.local(BaggageField.create(fieldName)));
+				}
+				List<String> remoteFields = tracingProperties.getBaggage().getRemoteFields();
+				for (String fieldName : remoteFields) {
+					factoryBuilder.add(BaggagePropagationConfig.SingleBaggageField.remote(BaggageField.create(fieldName)));
+				}
+				baggagePropagationCustomizers.ifAvailable(customizers -> customizers.forEach(customizer -> customizer.customize(factoryBuilder)));
+				return factoryBuilder.build();
+			}
+
+			/**
+			 * This has to be conditional as it creates a bean of type {@link SpanHandler}.
+			 *
+			 * <p>
+			 * {@link SpanHandler} beans, even if {@link SpanHandler#NOOP}, can trigger {@code SamplerCondition}
+			 */
+			@Configuration(proxyBeanMethods = false)
+			@Conditional(BaggageTagSpanHandlerCondition.class)
+			static class BaggageTagSpanHandlerConfiguration {
+
+				@Bean
+				SpanHandler baggageTagSpanHandler(TracingProperties properties) {
+					List<String> tagFields = properties.getBaggage().getTagFields();
+					if (tagFields.isEmpty()) {
+						return SpanHandler.NOOP; // Brave ignores these
+					}
+					return new BaggageTagSpanHandler(tagFields.stream().map(BaggageField::create).toArray(BaggageField[]::new));
+				}
+
+			}
 
+			/**
+			 * We need a special condition as it users could use either comma or yaml encoding,
+			 * possibly with a deprecated prefix.
```

Comment:
Why could there be a deprecated prefix?

Reply:
Most likely that's some ancient thing that I unnecessarily copied over from Sleuth. I'll remove it